ccm_web/client/src/components/ConfirmationTable.tsx

```diff
@@ -0,0 +1,186 @@
+  const [rowsPerPage, setRowsPerPage] = React.useState(5)
+
+  useEffect(() => {
+    setTableRows(props.grades.map(i => {
+      return { rowNumber: i.rowNumber, uniqname: i.uniqname, grade: i.grade }
+    }).sort((a, b) => (a.rowNumber < b.rowNumber ? -1 : 1))
+    )
+  }, [props.grades])
+
+  const renderTable = (): JSX.Element => {
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, tableRows.length - page * rowsPerPage)
+
+    const handleChangePage = (event: unknown, newPage: number): void => {
+      setPage(newPage)
+    }
+
+    const handleChangeRowsPerPage = (event: { target: { value: string } }): void => {
+      setRowsPerPage(parseInt(event.target.value, rowsPerPage))
+      setPage(0)
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function TablePaginationActions (props: { count: any, page: any, rowsPerPage: any, onChangePage: any }): JSX.Element {
+      const classes = useStyles()
+      const theme = useTheme()
+      const { count, page, rowsPerPage, onChangePage } = props
+
+      const handleFirstPageButtonClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+        onChangePage(event, 0)
+      }
+
+      const handleBackButtonClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+        onChangePage(event, page - 1)
+      }
+
+      const handleNextButtonClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+        onChangePage(event, props.page + 1)
+      }
+
+      const handleLastPageButtonClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+        onChangePage(event, Math.max(0, Math.ceil(count / rowsPerPage) - 1))
+      }
+
+      return (
+        <div className={classes.root}>
+          <IconButton
+            onClick={handleFirstPageButtonClick}
+            disabled={page === 0}
+            aria-label="first page"
+          >
+            {theme.direction === 'rtl' ? <LastPageIcon /> : <FirstPageIcon />}
+          </IconButton>
+          <IconButton onClick={handleBackButtonClick} disabled={page === 0} aria-label="previous page">
+            {theme.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
+          </IconButton>
+          <IconButton
+            onClick={handleNextButtonClick}
+            disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+            aria-label="next page"
+          >
+            {theme.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
+          </IconButton>
+          <IconButton
+            onClick={handleLastPageButtonClick}
+            disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+            aria-label="last page"
+          >
+            {theme.direction === 'rtl' ? <FirstPageIcon /> : <LastPageIcon />}
+          </IconButton>
+        </div>
+      )
+    }
```

Comment:
I didn't diff this with the other table `ValidationErrorTable`, but this looks pretty much the same as the pagination component there, right? We could probably extract this as a separate component that you use in both places, no?

Reply:
We probably could

Reply:
So yeah, I agree that a wholesale refactoring of the entire tables to reduce duplication isn't necessary at this stage, but if this pagination component is the same, I'd like to request you pull it out to its own file and use it in both places. DRY will help us keep things consistent and maintain the project over time. If it's actually not that simple, let me know, we can postpone then.

Reply:
Looks like you did it, awesome, thanks.